### PR TITLE
DORIS-1710: Respect the advertised packet size in the CEA708 decoder

### PR DESCRIPTION
**Pull request by szaboa**
Fix for CEA708 decoder failure [DORIS-1710](https://dicetech.atlassian.net/browse/DORIS-1710).

Root cause of the failure is that the packet data in a `DtvCcPacket` is less than advertised in the first byte of the data (`packetSize`). 

Based on the specs the number of packet data bytes should be `packetSize*2-1`. What happens in our case that while reading out the packet data we get an **early** `PACKET_START` (0x03) - [L232](https://github.com/DiceTechnology/ExoPlayer/blob/2bd31c3416492c9e33f2b99942545278dfae13df/library/extractor/src/main/java/com/google/android/exoplayer2/text/cea/Cea708Decoder.java#L232), meaning our current packet will be incomplete (size is less than advertised). For this case, the specs says:

> Some decoders may reset if syntactic elements are left incomplete at the end of a DTVCC Caption Channel Packet, but there is no requirement for decoders to reset. If incomplete CEA-708-E 27 syntactic elements remain, unexpected information may be displayed, and the decoder may be left in an indeterminate state.

The ExoPlayer implementation still attempts to process the data (see comment in [L295](https://github.com/DiceTechnology/ExoPlayer/blob/2bd31c3416492c9e33f2b99942545278dfae13df/library/extractor/src/main/java/com/google/android/exoplayer2/text/cea/Cea708Decoder.java#L295)) but when building up the service block (`captionChannelPacketData`) then it is using the manually calculated packet data length (`currentDtvCcPacket.currentIndex`) instead of the advertised one (`currentDtvCcPacket.packetSize`) - see [L314](https://github.com/DiceTechnology/ExoPlayer/blob/2bd31c3416492c9e33f2b99942545278dfae13df/library/extractor/src/main/java/com/google/android/exoplayer2/text/cea/Cea708Decoder.java#L314). This could cause an exception later, by trying to read more bits than it actually has (e.g. WWE live stream):

```
Caused by: java.lang.IllegalStateException
        at com.google.android.exoplayer2.util.Assertions.checkState(Assertions.java:84)
        at com.google.android.exoplayer2.util.ParsableBitArray.assertValidOffset(ParsableBitArray.java:341)
        at com.google.android.exoplayer2.util.ParsableBitArray.skipBits(ParsableBitArray.java:145)
        at com.google.android.exoplayer2.text.cea.Cea708Decoder.handleSetPenLocation(Cea708Decoder.java:721)
        at com.google.android.exoplayer2.text.cea.Cea708Decoder.handleC1Command(Cea708Decoder.java:503)
```

Other decoder implementations are using the advertised `packetSize` when creating the service block even when the packet is incomplete. See here:
 * shaka: https://github.com/shaka-project/shaka-player/blob/main/lib/cea/dtvcc_packet_builder.js#L57
 * comcast/caption-inspector: https://github.com/Comcast/caption-inspector/blob/8e278ac00e0b1c09778c475d7632d572c706c27b/src/xform/dtvcc_decode.c#L334

Doing the same in ExoPlayer solves the problem and closed captions can be seen on the on WWE Live streams.
Note: tested this change on a [sample stream](https://sample-videos-zyrkp2nj.s3-eu-west-1.amazonaws.com/nasa-live/master-cea708.m3u8) where the packet sizes are correct and haven't seen any side effects.

Screenshot 2023-03-06 at 20 53 40

[DORIS-1710]: https://dicetech.atlassian.net/browse/DORIS-1710?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: library/extractor/src/main/java/com/google/android/exoplayer2/text/cea/Cea708Decoder.java [hunk 1]
```diff
@@ -310,7 +310,9 @@ private void processCurrentPacket() {
     // exhausted. On each iteration we process a single service block. If the block has a service
     // number different to the currently selected service, then we skip it and continue with the
     // next service block.
-    captionChannelPacketData.reset(currentDtvCcPacket.packetData, currentDtvCcPacket.currentIndex);
+    // TODO Fix for https://dicetech.atlassian.net/browse/DORIS-1710.
+    captionChannelPacketData.reset(currentDtvCcPacket.packetData, (currentDtvCcPacket.packetSize * 2 - 1));
+    //captionChannelPacketData.reset(currentDtvCcPacket.packetData, currentDtvCcPacket.currentIndex);
```
Comment (markhomoki) [inline library/extractor/src/main/java/com/google/android/exoplayer2/text/cea/Cea708Decoder.java:316]: do we need this?

File: library/extractor/src/main/java/com/google/android/exoplayer2/text/cea/Cea708Decoder.java [hunk 2]
```diff
@@ -310,7 +310,9 @@ private void processCurrentPacket() {
     // exhausted. On each iteration we process a single service block. If the block has a service
     // number different to the currently selected service, then we skip it and continue with the
     // next service block.
-    captionChannelPacketData.reset(currentDtvCcPacket.packetData, currentDtvCcPacket.currentIndex);
+    // TODO Fix for https://dicetech.atlassian.net/browse/DORIS-1710.
```
Comment (markhomoki) [inline library/extractor/src/main/java/com/google/android/exoplayer2/text/cea/Cea708Decoder.java]: why TODO?

Comment (szaboa) [reply library/extractor/src/main/java/com/google/android/exoplayer2/text/cea/Cea708Decoder.java]: To watch out more when migrating to new exo versions, it stands out more (yellow highlight in AS). But yeah, sounds weird so I'll remove it.

File: library/extractor/src/main/java/com/google/android/exoplayer2/text/cea/Cea708Decoder.java (hunk 1 above)
Comment (szaboa) [reply library/extractor/src/main/java/com/google/android/exoplayer2/text/cea/Cea708Decoder.java:316]: Yes I would keep this, to see easily what's the difference between the original cea708 decoder and our forked decoder.

**Review by guoen21 — APPROVED** [review]

**Pull request merged**